app/models/agents/weibo_publish_agent.rb

```diff
@@ -65,17 +74,38 @@ def receive(incoming_events)
             'success' => false,
             'error' => e.message,
             'failed_tweet' => tweet_text,
+            'failed_pic' => pic_url,
             'agent_id' => event.agent_id,
             'event_id' => event.id
           }
         end
+        # you can't tweet too fast, give it a minute, i mean... 10 seconds
+        sleep 10 if incoming_events.length > 1
       end
     end
 
     def publish_tweet text
       weibo_client.statuses.update text
     end
 
+    def publish_tweet_with_pic text, pic
+      weibo_client.statuses.upload text, open(pic)
```

Comment:
I think using `open` here is safe because of `valid_image?`, so I'm going to merge. It might be better to use a Net::HTTP request here though, instead of open-uri, since if `valid_image?` had a bug, someone could convince your Huginn to upload a file on disk to weibo. Think we should change this?